oak/server/logging_node.cc

```diff
@@ -37,12 +37,13 @@ void LoggingNode::Run() {
       LOG(WARNING) << "Node termination requested";
       return;
     }
-    ReadResult result = channel->Read(INT_MAX);
+    ReadResult result = channel->Read(INT_MAX, INT_MAX);
```

Comment:
I know we discussed this, but it still confuses me. Have you thought about having a peek function that returns the size?

Reply:
I don't think a peek function would help, because C++ callers don't need it (they get ownership of the `Message`, so don't have any size concerns) and for Wasm callers (via `wasm::channel_read`) it would expand the syscall surface without adding any new functionality:
 - `wasm::channel_read` will still need to cope with the case that the provided byte/handle buffers are too small, because it can't rely on the node code doing things correctly
 - peeking at sizes can already be done with something like `wasm::channel_read(handle, nil, &size, nil, &count)`.

But you're right that it's a bit hard to follow &ndash; I've had a go at re-writing the `MessageChannelReadHalf::Read` comment.